lib/ansible/vars/__init__.py

```diff
@@ -578,15 +578,18 @@ def _load_inventory_file(self, path, loader):
         else:
             file_name, ext = os.path.splitext(path)
             data = None
-            if not ext or ext not in C.YAML_FILENAME_EXTENSIONS:
-                for test_ext in C.YAML_FILENAME_EXTENSIONS:
-                    new_path = path + test_ext
-                    if loader.path_exists(new_path):
-                        data = loader.load_from_file(new_path)
-                        break
-            else:
+            if not ext or ext in C.YAML_FILENAME_EXTENSIONS:
+                # if no extension or in valid extensions
                 if loader.path_exists(path):
                     data = loader.load_from_file(path)
+            else:
+                # try appending yaml extensions to find  valid file
+                for test_ext in C.YAML_FILENAME_EXTENSIONS:
+                    if test_ext: # avoid empty extension as that just adds all files
```

Comment:
he, i find the other way easier to understand .. but generator might be slight optimization in this case